### ISSUE TITLE
`break` and `continue` don't handle destructors correctly

### ISSUE THREAD

**Issue by Daniel-Cortez** [pawn-lang/compiler#531]
### Issue description:
In some situations `break` and `continue` may call the destructor for the same variable multiple times or not call it at all.

### Minimal complete verifiable example (MCVE):
```Pawn
stock operator~(Tag:values[], count)
{
    do {
        printf("%d", values[--count]);
    } while (count != 0);
}

main()
{
    new const bool:TRUE = true;
    do {
        new Tag:var = Tag:1;
        if (TRUE)
        {
            break;
        }
    } while (!TRUE);
}
```
The destructor for `var` is not called on `break`.

Another example, this time with `continue` (the destructor is the same as in the previous example):
```Pawn
main()
{
    new Tag:var = Tag:1;
    while (++var <= Tag:5)
        continue;
}
```
The destructor for `var` is called on every loop iteration, even though the variable is not defined inside the loop body.

### Workspace Information:

* Compiler version: 3.2.3664, 3.10.10
* Command line arguments provided (or sampctl version):
* Operating System:

**Comment by Daniel-Cortez** [comment]
The source of the problem lies in the concept of "compound statement nesting levels" and how the compiler handles it.
```Pawn
Func()
{
    // Compound level 1
    while (/* ... */)
    {
        // Level 2
        new Tag:var2 = Tag:1;
        if (/* ... */)
        {
            // Level 3
            new Tag:var3 = Tag:2;
            if (/* ... */)
            {
                // Level 4
                new Tag:var4 = Tag:3;
                break;
                // While we can see that the loop body is on compound level 2, the compiler
                // doesn't store this information anywhere, which is why "break" and "continue"
                // actually don't know for which levels exactly they should call destructors
                // and for which they shouldn't - instead they call destructors only for
                // the variables defined at the current level (in this case it's only "var4")
                // and perform a jump to the loop start/exit point, which means that for any
                // other variables defined on the previous levels inside the loop ("var2" and
                // "var3") the destructor is NOT called.
            }
        }
    }

    // We're back on level 1
    new Tag:var1;
    while (/* ... */)
        continue; // This is not a compound statement (there are no "{}"), so we're
                  // still on level 1 (yes, this is how the compiler works).
                  // "continue" calls destructors for variables defined at the current
                  // level, which means that the destructor for "var1" will be
                  // repeatedly called on each loop iteration, even though the variable
                  // is defined outside of the loop.

    // Level 1
    for (new Tag:i = Tag:0; ; ) // The "for" loop initialization section is counted as a
                                // separate level, which means that "i" is defined on level 2.
        continue; // This is still level 2, so "continue" calls destructors for this level,
                  // which is why the destructor for "i" is called at every iteration.
}
```
I think the best solution for this is to add a new field in the "loop table" entry struct to store the "compound statement" nesting level for the loop body. This will allow "break" and "continue" to actually know for which compound levels exactly they should call destructors, instead of doing so only for the current level.

**Comment by Daniel-Cortez** [comment]
Fixed in #532, closing.
